File: examples/hello-kotlin/functions/hello/Handler.kt
// hello — Flux function (compiled to WASM via Kotlin/Wasm)
// Build: ./gradlew wasmWasiJar
import kotlinx.serialization.json.*

@Suppress("unused")
@WasmExport("hello_handler")
fun HelloHandler(inputPtr: Int, inputLen: Int): Long {
    // TODO: decode input JSON at inputPtr
    val response = """{ "ok": true }"""
    val bytes    = response.encodeToByteArray()
    // Allocation: host is responsible for freeing.
    return (inputPtr.toLong() shl 32) or bytes.size.toLong()
}

File: examples/hello-swift/functions/hello/Handler.swift
// hello — Flux function (compiled to WASM via swiftwasm)
// Build: swiftc -target wasm32-unknown-wasi Handler.swift -o hello.wasm

@_cdecl("hello_handler")
func HelloHandler(inputPtr: UnsafePointer<UInt8>, inputLen: UInt32) -> UInt64 {
// TODO: decode JSON at inputPtr
let response = #"{\"ok\":true}"#
let bytes    = Array(response.utf8)
let outPtr   = UnsafeMutablePointer<UInt8>.allocate(capacity: bytes.count)
outPtr.initialize(from: bytes, count: bytes.count)
return (UInt64(UInt(bitPattern: outPtr)) << 32) | UInt64(bytes.count)
}
